packages/axios-extension/src/abap/lrep-service.ts

```diff
@@ -166,13 +166,20 @@ export class LayeredRepositoryService extends Axios implements Service {
      * Merge a given app descriptor variant with the stord app descriptor.
      *
      * @param appDescriptorVariant zip file containing an app descriptor variant
+     * @param workspacePath value for workspacePath URL parameter
      * @returns a promise with an object containing merged app descriptors with their id as keys.
      */
     public async mergeAppDescriptorVariant(
-        appDescriptorVariant: Buffer
+        appDescriptorVariant: Buffer,
+        workspacePath?: string
     ): Promise<{ [key: string]: MergedAppDescriptor }> {
+        let path = '/appdescr_variant_preview/';
+        if (workspacePath) {
+            path += `?workspacePath=${workspacePath}`;
```

Comment:
Technically this is not part of the path but it is a query parameter. While your approach works, it gets complicated if we start adding more optional parameters in the future, therefore, I'd always recommend handling parameters as parameters and not as concatenated strings, so please use the `params` property of the request config.

Reply:
The fix was a bit difficult here because passing the query parameter via the `params` automatically encodes the value which causes the request to fail with `Error 400 - Invalid workspace path`.  Another issue was the setting the `params` for the request overwrites the default params from the instance (where we have `saml2=disabled`).
I think with the following commits this should be fixed: cd4deddace55670fa261dc00d30444d9215d8b49 and 130d7a2cd8aecdd0df9ef124965390db316295f5

Reply:
Wow, interesting findings. I'd expect that the defaults are always added automatically.